src/settings/settings.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/sequelize';
+import { Settings } from './settings.model';
+
+@Injectable()
+export class SettingsService {
+  constructor(
+    @InjectModel(Settings)
+    private settingsRepository: typeof Settings,
+  ) {}
+
+  async createSetting(settingData: Settings): Promise<Settings> {
+    await this.validateDataType(settingData);
+    return await this.settingsRepository.create(settingData);
```

Comment:
just use `return` instead of `return await`. https://stackoverflow.com/a/42750371